R-packages/evalcast/tests/testthat/test-caching.R

```diff
@@ -0,0 +1,32 @@
+if(memoise::is.memoised(download_signal)) memoise::forget(download_signal)
+
+t1 = system.time(
+  suppressMessages(
+    get_predictions(
+      baseline_forecaster, "baby",
+      tibble::tibble(
+        data_source=c("jhu-csse", "usa-facts"),
+        signal = c("deaths_incidence_num","confirmed_incidence_num"),
+        start_day=lubridate::ymd("2020-09-15")),
+        lubridate::ymd("2020-10-01"),"epiweek", 1L, "state", "mi"
+    )
+  )
+)
+
+t2 = system.time(
+  suppressMessages(
+    get_predictions(
+      baseline_forecaster, "baby",
+      tibble::tibble(
+        data_source=c("jhu-csse", "usa-facts"),
+        signal = c("deaths_incidence_num","confirmed_incidence_num"),
+        start_day=lubridate::ymd("2020-09-15")),
+      lubridate::ymd("2020-10-01"),"epiweek", 1L, "state", "mi"
+    )
+  )
+)
+
+
+test_that("we are caching COVIDCAST calls", {
+  expect_true(t1[1] > t2[1])
```

Comment:
This does seem like a reasonable way to test that caching is working.  But I'm wondering if there could be cases where caching occurs but ends up being slower for some reason.  Is there a more direct way to verify that caching occurred? @bnaras, do you have thoughts on this?

Reply:
Also, perhaps you could add a test that the predictions cards from the two runs are identical?

Reply:
I added a test to check the prediction cards (both the attributes and the values). There's a note there about the attribute check for @bnaras and #98 . As for the timing check, I added a direct test to make sure that the caching is enabled. Waiting to push until I can try it locally.

Reply:
I pushed a fix to #144 so that you can try your tests.  I am not sure building the caching directly into the function is the right approach as that might screw up experiments where real-time changes may take place to the data. It seems it would be better if the user is given the option of using caching by means of a configuration setting...

Reply:
@jacobbien To answer your question about checking if caching occured,  if one uses the file system, there must be objects in the file system one can look at for the timestamp...

Reply:
I haven't been able to get it to work. There are two commented out tests that use `memoise::has_cache()` to actually examine the cache. One issue with the current approach (supporting @bnaras's point) is that the in-memory version is hard to access. But it greatly simplifies the package development. We could prompt on load to warn of the behavior and suggest caching locally instead. This `memoise` creates unintuitive file names though.